File: server/src/main/scala/pulsar_auth/conversions.scala
package pulsar_auth

import java.net.HttpCookie
import com.tools.teal.pulsar.ui.api.v1.pulsar_auth as pb

def credentialsTypePbFromCredentials(credentials: Credentials): pb.CredentialsType =
    credentials match
        case EmptyCredentials(_) => pb.CredentialsType.CREDENTIALS_TYPE_EMPTY
        case OAuth2Credentials(_, _, _, _, _) => pb.CredentialsType.CREDENTIALS_TYPE_OAUTH2
        case JwtCredentials(_, _) => pb.CredentialsType.CREDENTIALS_TYPE_JWT
        case AuthParamsStringCredentials(_, _, _) => pb.CredentialsType.CREDENTIALS_TYPE_AUTH_PARAMS_STRING

File: ui/components/app/pulsar-auth/conversions.ts
import * as pb from '../../../grpc-web/tools/teal/pulsar/ui/api/v1/pulsar_auth_pb';
import { MaskedCredentials, CredentialsType } from './domain';

export function maskedCredentialsFromPb(credentialsPb: pb.MaskedCredentials): MaskedCredentials {
  let type: CredentialsType;
  switch (credentialsPb.getType()) {
    case pb.CredentialsType.CREDENTIALS_TYPE_EMPTY:
      type = 'empty';
      break;
    case pb.CredentialsType.CREDENTIALS_TYPE_OAUTH2:
      type = 'oauth2';
      break;
    case pb.CredentialsType.CREDENTIALS_TYPE_JWT:
      type = 'jwt';
      break;
    default:
      throw new Error(`Unknown credentials type: ${credentialsPb.getType()}`);
  }

  return {
    name: credentialsPb.getName(),
    type,
  };
}
